### PR TITLE
Use go modules

### LINKED ISSUE #381: use go modules
/kind feature

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.]
per https://github.com/kubernetes-sigs/cluster-api/pull/1054/files
let's  plan to switch to go modules accordingly

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

### PR DESCRIPTION
**Pull request by hidekazuna**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #381 

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by hidekazuna** [comment]
I referred to https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/231

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+	k8s.io/apiextensions-apiserver v0.0.0-20190228180357-d002e88f6236 // indirect
+	k8s.io/apimachinery v0.0.0-20190612125636-6a5db36e93ad
+	k8s.io/client-go v0.0.0-20190228174230-b40b2a5939e4
+	k8s.io/cluster-bootstrap v0.0.0-20190612131323-aa3fd9f69a09
+	k8s.io/code-generator v0.0.0-20190612125529-c522cb6c26aa
+	k8s.io/component-base v0.0.0-20190615090122-7edc45fbd48e // indirect
+	k8s.io/gengo v0.0.0-20190327210449-e17681d19d3a // indirect
+	k8s.io/klog v0.3.3
+	k8s.io/kube-openapi v0.0.0-20190603182131-db7b694dc208 // indirect
+	sigs.k8s.io/cluster-api v0.1.4
```
Comment (jichenjc) [inline go.mod]: previously it's branch release-0.1 now it's a tag ? just want to make sure understand is correct

Comment (hidekazuna) [reply go.mod]: Yes, The go.mod file generally use semantics versions which is introduced by tagging a revision in the underlying source repository. 
https://golang.org/cmd/go/#hdr-Pseudo_versions

**Comment by jichenjc** [comment]
/approve

@sbueringer do you need this to be merged after #382 ?

**Comment by jichenjc** [comment]
I ran locally and able to create a cluster

**Comment by sbueringer** [comment]
@jichenjc if it's okay let's merge https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/401 then go modules. But I think we have consensus to merge the LB PR now so you can merge both now

**Comment by jichenjc** [comment]
@hidekazuna 

run this locally for 'make test' and saw additional changes like following, I guess you might need confirm this? 

```
+++ b/config/crds/openstackproviderconfig_v1alpha1_openstackclusterproviderspec.yaml
@@ -87,6 +87,10 @@ spec:
                       ipVersion:
                         format: int64
                         type: integer
+                      ipv6AddressMode:
+                        type: string
+                      ipv6RaMode:
+                        type: string
                       limit:
                         format: int64
                         type: integer
diff --git a/vendor/sigs.k8s.io/cluster-api/BUILD.bazel b/vendor/sigs.k8s.io/cluster-api/BUILD.bazel
deleted file mode 100644
index be252e22..00000000
--- a/vendor/sigs.k8s.io/cluster-api/BUILD.bazel
+++ /dev/null
@@ -1,21 +0,0 @@
-load("@bazel_gazelle//:def.bzl", "gazelle")
-load("//build:run_in_workspace_with_goroot.bzl", "workspace_binary")
-
-# gazelle:prefix sigs.k8s.io/cluster-api
-# gazelle:proto disable_global
-gazelle(
-    name = "gazelle",
-    external = "vendored",
-)
-
-workspace_binary(
-    name = "lint",
-    args = ["run"],
-    cmd = "@com_github_golangci_golangci-lint//cmd/golangci-lint",
-)
-
```

**Comment by hidekazuna** [comment]
> @hidekazuna
> 
> run this locally for 'make test' and saw additional changes like following, I guess you might need commit this?

@jichenjc I resolved conflict. check again, please.

**Pull request merged**